cupy/functional/piecewise.py

```diff
@@ -0,0 +1,84 @@
+import cupy
+
+from cupy import core
+
+_piecewise_krnl = core.ElementwiseKernel(
+    'U condlist, T funclist',
+    'T y',
+    ' if(condlist) y = funclist',
+    'piecewise_kernel'
+)
+
+
+def piecewise(x, condlist, funclist):
+    """Evaluate a piecewise-defined function.
+
+        Args:
+            x (cupy.ndarray): input domain
+            condlist (list of cupy.ndarray or bool scalars):
+                Each boolean array/ scalar corresponds to a function
+                in funclist. Length of funclist is equal to that of
+                condlist. If one extra function is given, it is used
+                as the default value when the otherwise condition is met
+            funclist (cupy.ndarray or list): list of scalar functions.
```

Comment:
NumPy does not guarantee support for `numpy.ndarray`.
You can support for `cupy.ndarray`, but we don't want to document it for compatibility.
```suggestion
            funclist (list of scalars): list of scalar functions.
```